dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/ClusterUtils.java

```diff
@@ -82,15 +81,9 @@ public static URL mergeUrl(URL remoteUrl, Map<String, String> localMap) {
             map.remove(DEFAULT_KEY_PREFIX + Constants.ASYNC_KEY);
 
             // remove method async entry.
-            Set<String> methodAsyncKey = new HashSet<>();
-            for (String key : map.keySet()) {
-                if (key != null && key.endsWith("." + Constants.ASYNC_KEY)) {
-                    methodAsyncKey.add(key);
-                }
-            }
-            for (String needRemove : methodAsyncKey) {
-                map.remove(needRemove);
-            }
+            map = map.entrySet().stream().filter(
+                    entry -> entry.getKey() == null || !entry.getKey().endsWith("." + Constants.ASYNC_KEY))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry:: getValue));

```

Comment:
The logic is right, but this may cause the code hard to read. Let see what other reviews say about this.

Reply:
A helper function in filter will be more concise to me.

Reply:
It's not hard to read if you are familiar with stream api of java
> The logic is right, but this may cause the code hard to read. Let see what other reviewers say about this.

Reply:
Yes you are right, I am considering this may confuses readers.. But it's okay then.